src/audio/eq_fir/eq_fir_ipc4.c

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2017 Intel Corporation. All rights reserved.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//         Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//         Keyon Jie <yang.jie@linux.intel.com>
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/data_blob.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/audio/ipc-config.h>
+#include <sof/common.h>
+#include <rtos/panic.h>
+#include <sof/ipc/msg.h>
+#include <rtos/alloc.h>
+#include <rtos/init.h>
+#include <sof/lib/memory.h>
+#include <sof/lib/uuid.h>
+#include <sof/list.h>
+#include <sof/math/fir_config.h>
+#include <sof/platform.h>
+#include <rtos/string.h>
+#include <sof/ut.h>
+#include <sof/trace/trace.h>
+#include <ipc/control.h>
+#include <ipc/stream.h>
+#include <ipc/topology.h>
+#include <kernel/abi.h>
+#include <user/eq.h>
+#include <user/fir.h>
+#include <user/trace.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
```

Comment:
something tells me that not all these headers are needed here. We do want to have all headers that are used in this file included explicitly, and it's difficult to keep the list perfect, but maybe you could do at least a light walk through and check if you find any obviously unused ones. E.g. uuid.h? platform.h? memory.h?

Reply:
that's correct, just need takes time